recipes/ginkgo/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+            del self.options.fPIC
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["GINKGO_BUILD_TESTS"] = False
+        self._cmake.definitions["GINKGO_BUILD_EXAMPLES"] = False
+        self._cmake.definitions["GINKGO_BUILD_BENCHMARKS"] = False
+        self._cmake.definitions["GINKGO_DEVEL_TOOLS"] = False
+        self._cmake.definitions["GINKGO_BUILD_REFERENCE"] = True
+        self._cmake.definitions["GINKGO_BUILD_OMP"] = self.options.openmp
+        self._cmake.definitions["GINKGO_BUILD_CUDA"] = self.options.cuda
+        self._cmake.definitions["GINKGO_BUILD_HIP"] = False
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        debug_suffix = "d" if self.settings.build_type == "Debug" else ""
+
+        self.cpp_info.components["reference"].libs = [
+            "ginkgo_reference" + debug_suffix]
+        self.cpp_info.components["cuda"].libs = ["ginkgo_cuda" + debug_suffix]
+        self.cpp_info.components["hip"].libs = ["ginkgo_hip" + debug_suffix]
+
+        self.cpp_info.components["omp"].libs = ["ginkgo_omp" + debug_suffix]
+        self.cpp_info.components["omp"].requires = ["cuda", "hip"]
+
+        self.cpp_info.components["core"].libs = ["ginkgo" + debug_suffix]
+        self.cpp_info.components["core"].requires = [
+            "reference", "omp", "cuda", "hip"]
```

Comment:
saw this PR after it was merged, but as usual my review for components: no generic conan components names when pkg_config name is not overridden for this component.

Reply:
What would this mean for the PR? Replacing `reference|cuda|omp|hip` by `ginkgo_reference|...`?

Reply:
Yes, qualify conan component names.

Reply:
I am sorry, I've added it in my notes... I'll pay more attention for it!